### PR TITLE
Evaluation fix

### PR DESCRIPTION
**Pull request by pdasigi**

### CODE REVIEW

File: src/main/java/org/allenai/wikitables/WikiTablesExample.java [hunk 1]
```diff
@@ -4,41 +4,85 @@
...
-    String superString = super.toString();
     List<String> logicalFormStrings = new ArrayList<>();
     for (Expression2 lf : logicalForms) {
         logicalFormStrings.add(lf.toString());
     }
-    return superString + " [\n" + String.join(" ", logicalFormStrings) + "\n]";
+    return sentence + " [\n" + String.join(" ", logicalFormStrings) + "\n]";
+  }
+
+  public static String toPnpLambdaForm(String expression) {
```
Comment (jayantk) [inline src/main/java/org/allenai/wikitables/WikiTablesExample.java]: Let's extract the logical form conversion methods to separate class containing only static utility functions, e.g., WikiTablesUtil.

Also, please don't do these conversions using regexes -- both of these methods are clearly broken if the variables have different names.

File: src/main/java/org/allenai/wikitables/WikiTablesExample.java [hunk 2]
```diff
@@ -4,41 +4,85 @@
...
+    expression = expression.replaceAll("lambda x", "lambda (x)");
+    expression = expression.replaceAll("(var x)", "x");
+    return expression;
+  }
+
+  public static String toSempreLambdaForm(String expression) {
+    // TODO: Properly de-canonicalize variable names, to work with multiple variables.
+    expression = expression.replaceAll("lambda \\(\\$0\\)", "lambda x");
+    // Remove single sub-expressions within parens. Eg: (fb:type.object.type (fb:type.row)) -> (fb:type.object.type fb:type.row)
+    expression = expression.replaceAll("\\(([^ ]*)\\)", "$1");
```
Comment (jayantk) [inline src/main/java/org/allenai/wikitables/WikiTablesExample.java]: This conversion is incorrect. (fb:type.row) is not equivalent to fb:type.row. One of these presumably shouldn't be generated by the parser, and the problem should be fixed there and not here.

File: src/main/scala/org/allenai/pnp/semparse/SemanticParser.scala [hunk 3]
```diff
@@ -368,7 +368,7 @@ class SemanticParser(val actionSpace: ActionSpace, val vocab: IndexedList[String
         currentScope.getVariableTemplates(curType) ++
         entityLinking.getEntitiesWithType(curType).map(_.template)
       val matches = templates.filter(_.matches(expIndex, exp, typeMap))
-      if (matches.size != 1) {
+      if (matches.size < 1) {
```
Comment (jayantk) [inline src/main/scala/org/allenai/pnp/semparse/SemanticParser.scala]: This is wrong. matches.size must == 1. If it doesn't, there's something wrong with the logical forms or the action space.

File: src/main/java/org/allenai/wikitables/WikiTablesExample.java (hunk 1 above)
Comment (jayantk) [inline src/main/java/org/allenai/wikitables/WikiTablesExample.java]: Also please rename these methods to "toPnpLogicalForm" and "toSempreLogicalForm". I'd also be fine if you abbreviated LogicalForm to Lf.

File: src/main/scala/org/allenai/wikitables/WikiTablesSemanticParserCli.scala [hunk 4]
```diff
@@ -326,12 +328,12 @@ class WikiTablesSemanticParserCli extends AbstractCli() {
           
       val beam = results.executions.slice(0, 10)
       val correct = beam.map { x =>
-        val simplified = simplifier.apply(x.value.decodeExpression)
```
Comment (jayantk) [inline src/main/scala/org/allenai/wikitables/WikiTablesSemanticParserCli.scala]: Why did you get rid of the simplifier? The simplifier just makes logical forms easier to read. Its output is semantically equivalent to its input.

**Comment by pdasigi** [comment]
Comments addressed in #39

**Comment by jayantk** [comment]
This PR got moved to the wikitables2 branch.

**Pull request closed without merging**